src/sentry/api/endpoints/organization_stats_v2.py

```diff
@@ -1,23 +1,125 @@
 
 import sentry_sdk
+from drf_spectacular.utils import OpenApiExample, extend_schema
+from rest_framework import serializers
 from rest_framework.exceptions import ParseError
 from rest_framework.request import Request
 from rest_framework.response import Response
+from typing_extensions import TypedDict
 
 from sentry.api.bases import NoProjects, OrganizationEventsEndpointBase
+from sentry.apidocs.constants import RESPONSE_NOTFOUND, RESPONSE_UNAUTHORIZED
+from sentry.apidocs.decorators import public
+from sentry.apidocs.parameters import GLOBAL_PARAMS
+from sentry.apidocs.utils import inline_sentry_response_serializer
 from sentry.constants import ALL_ACCESS_PROJECTS
 from sentry.search.utils import InvalidQuery
 from sentry.snuba.outcomes import (
+    COLUMN_MAP,
+    GROUPBY_MAP,
     QueryDefinition,
     massage_outcomes_result,
     run_outcomes_query_timeseries,
     run_outcomes_query_totals,
 )
 from sentry.snuba.sessions_v2 import InvalidField, InvalidParams
 from sentry.types.ratelimit import RateLimit, RateLimitCategory
+from sentry.utils.outcomes import Outcome
 
 
+class OrgStatsQueryParamsSerializer(serializers.Serializer):
+    # TODO: refactor endpoint and use this to validate query parameters.
+    # TODO: define fields for date parameters for use elsewhere
+    # time params
+    statsPeriod = serializers.CharField(
+        help_text=(
+            "This defines the range of the time series, relative to now. "
+            "The range is given in a `<number><unit>` format. "
+            "For example `1d` for a one day range. Possible units are `m` for minutes, `h` for hours, `d` for days and `w` for weeks."
+        ),
+        required=False,
+    )
+    interval = serializers.CharField(
+        help_text=(
+            "This is the resolution of the time series, given in the same format as `statsPeriod`. "
+            "The default resolution is `1h` and the minimum resolution is currently restricted to `1h` as well. "
+            "Intervals larger than `1d` are not supported, and the interval has to cleanly divide one day."
+        ),
+        required=False,
+    )
+    start = serializers.DateTimeField(
+        help_text="This defines the start of the time series range as an explicit datetime.",
+        required=False,
+    )
+    end = serializers.DateTimeField(
+        help_text="This defines the end of the time series range as an explicit datetime.",
+        required=False,
+    )
+
+    groupBy = serializers.MultipleChoiceField(
+        list(GROUPBY_MAP.keys()),
+        required=True,
+        help_text=(
+            "can pass multiple groupBy parameters to group by multiple, e.g. `groupBy=project&groupBy=outcome` to group by multiple dimensions. "
+            "Note that grouping by project can cause missing rows if the number of projects / interval is large. "
+            "If you have a large number of projects, we reccomend filtering and querying by them individually."
```

Comment:
```suggestion
            "If you have a large number of projects, we recommend filtering and querying by them individually."
```

Reply:
have a spell checker installed in my IDE going forward lol 🙈  thanks for the catch